File: src/main/scala/com/jimi/scala/KafkaDemo/kafkaConsumer.scala
package com.jimi.scala.KafkaDemo

import java.util.Properties

import org.apache.flink.api.common.serialization.SimpleStringSchema
import org.apache.flink.streaming.api.TimeCharacteristic
import org.apache.flink.streaming.api.scala.{DataStream, StreamExecutionEnvironment}
import org.apache.flink.streaming.connectors.kafka.FlinkKafkaConsumer

/**
  * @Author Administrator
  * @create 2019/8/23 15:11
  */

object kafkaConsumer {
  def main(args: Array[String]): Unit = {

    val env = StreamExecutionEnvironment.getExecutionEnvironment

    env.setStreamTimeCharacteristic(TimeCharacteristic.ProcessingTime)

    val properties = new Properties()
    properties.setProperty("bootstrap.servers","master126:9092")
//    properties.setProperty("zookeeper.connect","master126:2181")
    val topic = "flink-topic"
    properties.setProperty("group.id", "cloudera_mirrormaker")

    val kafkStream: FlinkKafkaConsumer[String] = new FlinkKafkaConsumer(topic,new SimpleStringSchema(),properties)
    kafkStream.setStartFromLatest()

    import org.apache.flink.api.scala._
    val stream: DataStream[String] = env.addSource(kafkStream)

    stream.print()

    env.execute("kafkaStreamPrintln")

  }
}

File: src/main/java/com/jimi/java/kafkademo/KafkaConsumer.java
package com.jimi.java.kafkademo;

import org.apache.flink.api.common.serialization.SimpleStringSchema;
import org.apache.flink.streaming.api.datastream.DataStream;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
import org.apache.flink.streaming.connectors.kafka.FlinkKafkaConsumer;

import java.util.Properties;

public class KafkaConsumer {

    public static void main(String[] args) throws Exception{

        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
        Properties properties = new Properties();
        properties.setProperty("bootstrap.servers", "master126:9092");

        FlinkKafkaConsumer<String> consumer = new FlinkKafkaConsumer("flink-topic", new SimpleStringSchema(), properties);

        // 从最早开始消费
        consumer.setStartFromLatest();
//        consumer.setStartFromEarliest();
        DataStream<String> stream = env.addSource(consumer);
        stream.print();
        // stream.map();
        env.execute();
    }
}//
